x-pack/plugins/monitoring/server/lib/setup/collection/get_collection_status.js

```diff
@@ -0,0 +1,365 @@
+                by_timestamp: {
+                  max: {
+                    field: 'timestamp'
+                  }
+                },
+                beat_type: {
+                  terms: {
+                    field: 'beats_stats.beat.type'
+                  }
+                }
+              }
+            },
+            logstash_uuids: {
+              terms: {
+                field: 'logstash_stats.logstash.uuid'
+              },
+              aggs: {
+                by_timestamp: {
+                  max: {
+                    field: 'timestamp'
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  };
+
+  const { callWithRequest } = req.server.plugins.elasticsearch.getCluster('monitoring');
+  return await callWithRequest(req, 'search', params);
+};
+
+async function detectProducts(req) {
+  const result = {
+    [KIBANA_SYSTEM_ID]: {
+      doesExist: true,
+    },
+    'elasticsearch': {
+      doesExist: true,
+    },
+    [BEATS_SYSTEM_ID]: {
+      mightExist: false,
+    },
+    [APM_CUSTOM_ID]: {
+      mightExist: false,
+    },
+    [LOGSTASH_SYSTEM_ID]: {
+      mightExist: false,
+    }
+  };
+
+  const msearch = [
+    { index: '*beat-*' },
+    { size: 0, terminate_after: 0 },
+
+    { index: '.management-beats*' },
+    { size: 0, terminate_after: 0 },
+
+    { index: 'logstash-*' },
+    { size: 0, terminate_after: 0 },
+
+    { index: '.logstash*' },
+    { size: 0, terminate_after: 0 },
+
+    { index: 'apm*' },
+    { size: 0, terminate_after: 0 },
+  ];
+
+  const { callWithRequest } = req.server.plugins.elasticsearch.getCluster('monitoring');
+  const {
+    responses: [
+      beatsDetectionResponse1,
+      beatsDetectionResponse2,
+      logstashDetectionResponse1,
+      logstashDetectionResponse2,
+      apmDetectionResponse
+    ]
+  } = await callWithRequest(req, 'msearch', { body: msearch });
+
+  if (get(beatsDetectionResponse1, 'hits.total.value', 0) > 0 || get(beatsDetectionResponse2, 'hits.total.value', 0) > 0) {
+    result[BEATS_SYSTEM_ID].mightExist = true;
+  }
+
+  if (get(logstashDetectionResponse1, 'hits.total.value', 0) > 0 || get(logstashDetectionResponse2, 'hits.total.value', 0) > 0) {
+    result[LOGSTASH_SYSTEM_ID].mightExist = true;
+  }
+
+  if (get(apmDetectionResponse, 'hits.total.value', 0) > 0) {
+    result[APM_CUSTOM_ID].mightExist = true;
+  }
+
+  return result;
+}
+
+function getUuidBucketName(productName) {
+  switch (productName) {
+    case 'elasticsearch':
+      return 'es_uuids';
+    case KIBANA_SYSTEM_ID:
+      return 'kibana_uuids';
+    case BEATS_SYSTEM_ID:
+    case APM_CUSTOM_ID:
+      return 'beats_uuids';
+    case LOGSTASH_SYSTEM_ID:
+      return 'logstash_uuids';
+  }
+}
+
+function isBeatFromAPM(bucket) {
+  if (!bucket.beat_type) {
+    return false;
+  }
+
+  return get(bucket, 'beat_type.buckets[0].key') === 'apm-server';
+}
+
+function shouldSkipBucket(product, bucket) {
```

Comment:
It's not super clear to me _why_ the bucket should be skipped. Consider either renaming this function to clarify its intent via its name OR if that's not practical, then add a comment above it.